### DISCUSSION TITLE
Concatenation of 2 strings in variables

### DISCUSSION THREAD

**Discussion by github-reader** [hairyhenderson/gomplate#1168] [Q&A] [answered]
Hello I want to do this:
```
{{ $var1 := "val1" }}
{{ $var2 := "val2" }}
/* Wanted result, no valid gomplate code */
{{ $var3 := $var1 + $var2 }}
```
I have googled for a solution. It seems that gomplate cannot concatenate strings. How ca I solve that with gomplate? Hints to go language did not help. They also did not work here?

**Comment by hairyhenderson** [comment, marked as answer]
Hi @github-reader, I've moved this issue to a discussion instead as this is more of a question than a bug report.

There are a number of ways to concatenate strings in the Go templating language that gomplate uses. Usually, what I do is use the `print` keyword:

```console
$ gomplate -i '{{ $var1 := "val1" }}{{ $var2 := "val2" }}{{ $var3 := print $var1 $var2 }}{{ $var3 }}'
val1val2
```

You can also use `printf` if you're familiar with that style of syntax:

```console
$ gomplate -i '{{ $var1 := "val1" }}{{ $var2 := "val2" }}{{ $var3 := printf "%s%s" $var1 $var2 }}{{ $var3 }}'
val1val2
```
